axelrod/tournament.py

```diff
@@ -88,57 +88,41 @@ def _build_result_set(self):
         -------
         axelrod.ResultSet
         """
-        result_set = ResultSet(
-            players=self.players,
-            interactions=self.interactions,
+        result_set = ResultSetFromFile(
+            filename=self.filename,
             with_morality=self._with_morality)
+        self.outputfile.close()
         return result_set
 
-    def _build_cache_required(self):
-        """
-        A boolean to indicate whether it is necessary to build the
-        deterministic cache.
-        """
-        return (
-            not self.noise and len(self.deterministic_cache) == 0)
-
-    def _build_cache(self, matches):
-        """
-        For parallel processing, this runs a single round robin in order to
-        build the deterministic cache.
-
-        Parameters
-        ----------
-        matches : list
-            The list of matches to update
-        """
-        self._logger.debug('Playing first round robin to build cache')
-        self._run_single_repetition(matches)
-        self._parallel_repetitions -= 1
-
-    def _run_single_repetition(self, interactions):
-        """
-        Runs a single round robin and updates the matches list.
-        """
-        new_matches = self.match_generator.build_matches(noise=self.noise)
-        interactions = self._play_matches(new_matches)
-        self.interactions.append(interactions)
-
-    def _run_serial_repetitions(self, interactions):
+    def _run_serial(self):
         """
         Runs all repetitions of the round robin in serial.
-
-        Parameters
-        ----------
-        ineractions : list
-            The list of interactions per repetition to update with results
         """
-        self._logger.debug('Playing %d round robins' % self.repetitions)
-        for repetition in range(self.repetitions):
-            self._run_single_repetition(interactions)
+        chunks = self.match_generator.build_match_chunks()
+
+        for chunk in chunks:
+            interactions = self._play_matches(chunk)
+            self._write_interactions(interactions)
         return True
 
-    def _run_parallel_repetitions(self, interactions):
+    def _write_interactions(self, interactions):
+        """Either write to memory or to file"""
```

Comment:
I took it out since it's just as fast to write to disk and it seemed more simple to just use a temp file instead. I think if we have "build_results" as suggested above then maybe we don't need this.